test/src/main/java/org/apache/accumulo/test/CompactionIT.java

```diff
@@ -674,48 +681,96 @@ public fooSelectorException(String message) {
         }
     }
 
+    public static class FailureListener extends RunListener {
+
+        private RunNotifier runNotifier;
+
+        public FailureListener(RunNotifier runNotifier) {
+            super();
+            this.runNotifier=runNotifier;
+        }
+
+        @Override
+        public void testFailure(Failure failure) throws Exception {
+            super.testFailure(failure);
+            this.runNotifier.pleaseStop();
+        }
+    }
+
+        public static class Retry implements TestRule {
+            private int retryCount;
+
+            public Retry(int retryCount) {
+                this.retryCount = retryCount;
+            }
+
+            public Statement apply(Statement base, Description description) {
+                return statement(base, description);
+            }
+
+            private Statement statement(final Statement base, final Description description) {
+                return new Statement() {
+                    @Override
+                    public void evaluate() throws Throwable {
+                        Throwable caughtThrowable = null;
+
+                        // implement retry logic here
+                        for (int i = 0; i < retryCount; i++) {
+                            try {
+                                base.evaluate();
+                                return;
+                            } catch (Throwable t) {
+                                caughtThrowable = t;
+                                System.err.println(description.getDisplayName() + ": run " + (i+1) + " failed");
+                            }
+                        }
+                        System.err.println(description.getDisplayName() + ": giving up after " + retryCount + " failures");
+                        throw caughtThrowable;
+                    }
+                };
+            }
+        }
+
+    @Rule
```

Comment:
Is there a way to do this without using a junit rule, or structure it to be closer to the new junit 5 @ExtendsWith annotation?  I believe that rules are one area that changed in JUnit5, but I am unclear on the details.  

This is not a deal breaker, just asking so that when we migrate to Junit 5, it would be one less thing that needed to migrate.  Certainly not the only place that we'd run into issues, just speculating that if we keep Junit5 in mind going forward, it may reduce the migrations required and make a change over easier.

Reply:
I don't think we want to retry in the test at all. What we probably want to do is have the test verify that Accumulo itself retries (or maybe fails gracefully? I'm not sure the best behavior) in the face of a compaction selector failure. I spoke with @tynyttie about this, and he's going to take another look. In the meantime, I'm going to mark this PR as "Draft" for now.